src/nrnpython/nrnpy_hoc.cpp

```diff
@@ -130,6 +130,64 @@ PyTypeObject* hocobject_type;
 
 static PyObject* hocobj_call(PyHocObject* self, PyObject* args, PyObject* kwrds);
 
+struct hocclass {
+    PyTypeObject head;
+};
+
+static int hocclass_init(hocclass* cls, PyObject* args, PyObject* kwds) {
+    if (PyType_Type.tp_init((PyObject*) cls, args, kwds) < 0) {
+        return -1;
+    }
+    return 0;
+}
+
+static PyObject* pytype_getname(PyTypeObject* pto) {
+#if PY_VERSION_HEX >= 0x030B0000  // since python-3.11
+    return PyType_GetName(pto);
+#else
+    return PyObject_GetAttrString((PyObject*) pto, "__name__");
+#endif
+}
+
+static PyObject* hocclass_getitem(PyObject* self, Py_ssize_t ix) {
+    assert(PyType_Check(self));
+    auto uname = pytype_getname((PyTypeObject*) self);
+    assert(uname);
+    const char* name = PyUnicode_AsUTF8AndSize(uname, 0);
+    assert(name);
+    Symbol* sym = hoc_table_lookup(name, hoc_built_in_symlist);
+    assert(sym);
+
+    assert(sym->type == TEMPLATE);
+    hoc_Item *q, *ql = sym->u.ctemplate->olist;
+    Object* ob;
+    ITERATE(q, ql) {
+        ob = OBJ(q);
+        if (ob->index == ix) {
+            return nrnpy_ho2po(ob);
+        }
+    }
+    char e[200];
+    Sprintf(e, "%s[%ld] instance does not exist", sym->name, ix);
+    PyErr_SetString(PyExc_IndexError, e);
+    return NULL;
+}
+
+// Note use of slots was informed by nanobind (search for nb_meta)
+
+static PyType_Slot hocclass_slots[] = {{Py_tp_base, nullptr},  // &PyType_Type : not obvious why it
+                                                               // must be set at runtime
+                                       {Py_tp_init, (void*) hocclass_init},
+                                       {Py_sq_item, (void*) hocclass_getitem},
+                                       {0, NULL}};
+
+static PyType_Spec hocclass_spec = {.name = "hoc.HocClass",
+                                    .basicsize = 0,
+                                    .itemsize = 0,
+                                    .flags = Py_TPFLAGS_DEFAULT,  //  | Py_TPFLAGS_HEAPTYPE,
+                                    .slots = hocclass_slots};
```

Comment:
> ```T object = { .des1 = arg1 , .des2 { arg2 } ... };	(3)	(since C++20)```

This is really surprising to me. I see that there are many ```static PyType_Spec ...``` examples in nrnpy_nrn.h of the form ```T object = { arg1, arg2, ... };```
To avoid the C++20 requirement, I'll just remove the descriptor tokens.